### PR TITLE
getBanks() optional params for filtering

### PR DESCRIPTION
**Pull request by zixunjian**

### CODE REVIEW

File: src/http/HttpClient.js [hunk 1]
```diff
@@ -203,12 +203,30 @@ class HttpClient {
...
+        let baseUrl = `/banks?`;
+        for (const id of ids) {
+            baseUrl += `ids=${id}&`;
+        }
+        if (page) baseUrl += `page=${page}&`;
+        if (perPage) baseUrl += `perPage=${perPage}&`;
         const request = {
             method: 'get',
-            url: `/banks`
+            url: `${baseUrl}search=${search}&country=${country}`,
```
Comment (yuhanniu) [inline src/http/HttpClient.js]: It seems that you append search and country even when they are empty strings?

**Comment by yuhanniu** [comment]
You might want to url encode the parameters

File: src/http/HttpClient.js (hunk 1 above)
Comment (zixunjian) [reply src/http/HttpClient.js]: yeah it doesnt affect the results

File: src/main/Token.js [hunk 2]
```diff
@@ -314,11 +314,12 @@ class Token {
     /**
      * Gets a list of available banks for linking
      *
+     * @param {Object} options - optional parameters for getBanks
      * @return {Promise} banks - list of banks
      */
-    getBanks() {
+    getBanks(options) {
```
Comment (mariano54) [inline src/main/Token.js]: Why are we using options instead of passing each separately?

Comment (zixunjian) [reply src/main/Token.js]: too many params imo, you think it's better that way?

File: src/http/HttpClient.js (hunk 1 above)
Comment (yuhanniu) [reply src/http/HttpClient.js]: yeah but that seems a little bit hacky. Since this is sdk I suggest that we do the right thing

Comment (zixunjian) [reply src/http/HttpClient.js]: sure

File: src/main/Token.js (hunk 2 above)
Comment (mariano54) [reply src/main/Token.js]: No, just wondering the rationale, makes sense

**Pull request merged**